cdap-gateway/src/main/java/co/cask/cdap/gateway/router/RouterPathLookup.java

```diff
@@ -83,8 +89,12 @@ private RouteDestination getV3RoutingService(String [] uriParts, AllowedMethod r
       //       <user-defined-method-path>"
       String serviceName = ServiceDiscoverable.getName(uriParts[2], uriParts[4], uriParts[8]);
       String version = uriParts[6];
-      // TODO: Optimize this instead of creating new object every time
-      return new RouteDestination(serviceName, version);
+      MultiKey cacheKey = new MultiKey(serviceName, version);
```

Comment:
This is not an optimization at all. It used to creating one `RouteDestination` object. Now it is creation of one `MultiKey`, plus two lookup in a concurrent map (`containsKey` and `get`). I don't think we need to change anything here. Simply remove the TODO and mark the CDAP-8014 as won't fix.

Shouldn't do any optimization without actual perf test backing it.